MidiBtlePairing/src/main/java/com/mobileer/example/midibtlepairing/MainActivity.java

```diff
@@ -48,16 +48,26 @@
 import java.util.HashMap;
 
 /*
- * App that provides a MIDI echo service.
+ * App that creates a Device for a BLE-MIDI peripheral.
+ * The peripheral can then be used by other apps.
  */
 public class MainActivity extends Activity {
     private static final String TAG = "MidiBtlePairing";
     private static final String PACKAGE_NAME = "com.mobileer.example.midibtlepairing";
 
     private MidiManager mMidiManager;
     private OpenDeviceListAdapter mOpenDeviceListAdapter;
+    private static final int REQUEST_ID_MULTIPLE_PERMISSIONS = 100; // arbitrary
```

Comment:
I'm a little confused about the arbituary request code then checking for that later. Why do we need this?

Reply:
It is used to match the call to onRequestPermissionsResult() with the request.  The may be multiple requests in progress.

https://developer.android.com/training/permissions/requesting#manage-request-code-yourself